aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3774,6 +4244,95 @@ Tensor _det_lu_based_helper_backward_helper(
   }
 }
 
+DEFINE_DISPATCH(ldl_factor_stub);
+
+TORCH_IMPL_FUNC(linalg_ldl_factor_ex_out)
+(const Tensor& self,
+ bool hermitian,
+ bool check_errors,
+ const Tensor& LD,
+ const Tensor& pivots,
+ const Tensor& info) {
+  // LAPACK workspace query segfalts if the input has 0 in batch dimensions.
+  if (self.numel() == 0) {
+    info.zero_();
+    return;
+  }
+
+  auto pivots_ = pivots.expect_contiguous();
+  auto info_ = info.expect_contiguous();
+
+  auto LD_ = at::native::borrow_else_clone(
+      LD.mT().is_contiguous(), LD, self, /*row_major=*/false);
+  if (LD.mT().is_contiguous()) {
+    LD_->copy_(self);
+  }
+
+  // We decided not to include upper flag in the API.
+  // https://github.com/pytorch/pytorch/pull/69828#issuecomment-1015143819
+  // We can revisit this decision later and remove upper completely
+  // also from low level functions or add it to the public API.
+  bool upper = false;
+  if (upper) {
+    LD_->triu_();
+  } else {
+    LD_->tril_();
+  }
+
+  // call ldl_factor_stub that fills the result tensors
+  ldl_factor_stub(
+      self.device().type(), *LD_, *pivots_, *info_, upper, hermitian);
+
+  if (!LD.is_same(*LD_)) {
+    LD.copy_(*LD_);
+  }
+  if (!info.is_same(*info_)) {
+    info.copy_(*info_);
+  }
+  if (!pivots.is_same(*pivots_)) {
+    pivots.copy_(*pivots_);
+  }
+
+  if (check_errors) {
+    at::_linalg_check_errors(
+        info, "torch.linalg.ldl_factor_ex", self.dim() == 2);
+  }
+}
+
+DEFINE_DISPATCH(ldl_solve_stub);
+
+TORCH_IMPL_FUNC(linalg_ldl_solve_out)
+(const Tensor& LD,
+ const Tensor& pivots,
+ const Tensor& B,
+ bool hermitian,
+ const Tensor& result) {
+  if (LD.numel() == 0 || pivots.numel() == 0) {
+    return;
+  }
+
+  Tensor B_broadcast;
+  std::tie(B_broadcast, std::ignore) =
+      _linalg_broadcast_batch_dims(B, LD, /*don't check errors*/ nullptr);
```

Comment:
Is this now just equivalent to `auto B_broadcast = B.expand_as(result);`?

Reply:
Right. I removed it now because we don't need to expand `B` as `result` to call `result.copy_(B)`.